src/storage/storage.ts

```diff
@@ -0,0 +1,809 @@
+   */
+  async editToken(tokenUid: string, meta: Partial<ITokenMetadata>): Promise<void> {
+    this.store.editToken(tokenUid, meta);
+  }
+
+  /**
+   * Iterate on all tokens on the storage.
+   *
+   * @returns {AsyncGenerator<ITokenData & Partial<ITokenMetadata>>}
+   */
+  async *getAllTokens(): AsyncGenerator<ITokenData & Partial<ITokenMetadata>> {
+    for await (const token of this.store.tokenIter()) {
+      yield token;
+    }
+  }
+
+  /**
+   * Iterate on all registered tokens of the wallet.
+   *
+   * @returns {AsyncGenerator<ITokenData & Partial<ITokenMetadata>>}
+   */
+  async *getRegisteredTokens(): AsyncGenerator<ITokenData & Partial<ITokenMetadata>> {
+    for await (const token of this.store.registeredTokenIter()) {
+      yield token;
+    }
+  }
+
+  /**
+   * Get a token from storage along with the metadata of the wallet transactions.
+   *
+   * @param {string|undefined} [token='00'] Token uid to fetch
+   * @returns {Promise<(ITokenData & Partial<ITokenMetadata>)|null>}
+   */
+  async getToken(token?: string): Promise<(ITokenData & Partial<ITokenMetadata>)|null> {
+    return this.store.getToken(token || HATHOR_TOKEN_CONFIG.uid);
+  }
+
+  /**
+   * Regsiter a token.
+   * @param {ITokenData} token Token data to register
+   * @returns {Promise<void>}
+   */
+  async registerToken(token: ITokenData): Promise<void> {
+    await this.store.registerToken(token);
+  }
+
+  /**
+   * Unregister a token from the wallet.
+   * @param {Promise<void>} tokenUid Token uid to unregister.
+   * @returns {Promise<void>}
+   */
+  async unregisterToken(tokenUid: string): Promise<void> {
+    await this.store.unregisterToken(tokenUid);
+  }
+
+  /**
+   * Iterate on all utxos of the wallet.
+   * @returns {AsyncGenerator<IUtxo, any, unknown>}
+   */
+  async *getAllUtxos(): AsyncGenerator<IUtxo, any, unknown> {
+    for await (const utxo of this.store.utxoIter()) {
+      yield utxo;
+    }
+  }
+
+  /**
+   * Select utxos matching the request and do not select any utxos marked for inputs.
+   *
+   * @param {Omit<IUtxoFilterOptions, 'reward_lock'>} [options={}] Options to filter utxos and stop when the target is found.
+   * @returns {AsyncGenerator<IUtxo, any, unknown>}
+   */
+  async *selectUtxos(options: Omit<IUtxoFilterOptions, 'reward_lock'> = {}): AsyncGenerator<IUtxo, any, unknown> {
+    const newFilter = (utxo: IUtxo): boolean => {
+      const utxoId = `${utxo.txId}:${utxo.index}`;
+      return (!this.utxosSelectedAsInput.has(utxoId)) && (options.filter_method ? options.filter_method(utxo) : true);
+    }
+
+    const newOptions: IUtxoFilterOptions = {
+      ...options,
+      filter_method: newFilter,
+    };
+    if (this.version?.reward_spend_min_blocks) {
+      newOptions.reward_lock = this.version.reward_spend_min_blocks;
+    }
+    for await (const utxo of this.store.selectUtxos(newOptions)) {
+      yield utxo;
+    }
+  }
+
+  /**
+   * Calculate the token balance of a transaction, including authorities.
```

Comment:
As stated above this was created during the review process, it was part of `fillTx` where it made sense.
How it worked before:

- Any output is added to the balance
  - The `fillTx` method will be responsible for finding inputs to match the outputs.
- Only inputs from the wallet are added
  - The old way would only consider inputs of txs on storage so the behavior is kept
  - This is bad for atomic-swaps but since inputs are not selected with this method it should not be an issue for now.

With this being said, I choose the name very poorly for what it is actually doing.

Reply:
The method has been removed.